build-in-container/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM mcr.microsoft.com/onebranch/cbl-mariner/build:2.0
+
+RUN mkdir -m 777 -p /sources && \
+    for i in $(seq 1 12); do \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/dev/pts; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/dev/null; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/proc; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/sys; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/run; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/localrpms/noarch; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/localrpms/aarch64; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/localrpms/x86_64; \
+                mkdir -m 777 -p /temp/DockerStage/docker-chroot-$i/localrpms/upstream-cached-rpms; \
+    done
+
+RUN tdnf update -qy tdnf && \
+        tdnf -y install \
+                binutils \
+                bison \
+                cdrkit \
+                curl \
+                dnf \
+                dnf-utils \
+                dosfstools \
+                gawk \
+                gcc \
+                git \
+                glibc-devel \
+                golang-1.17.13-1.cm2 \
```

Comment:
Do we need to have this strict dependency on golang-1.17?
We recently confirmed that our latest supplied golang will work for our tools - https://github.com/microsoft/CBL-Mariner/pull/5630

Consider verifying with the latest golang and see if we can use that, rather than the pinned version.

Reply:
I had the same question. Let me try to build without explicit version like you suggested, thanks